### PR TITLE
Delete extra services on rollback/restore

### PR DESCRIPTION
**Pull request by jafcrocker**
Fixes https://jira.zenoss.com/browse/ZEN-15903
Enable code to delete services which are not part of the backup
Removed dead code

### CODE REVIEW

File: dfs/backup.go [hunk 1]
```diff
@@ -532,8 +532,10 @@ func (dfs *DistributedFilesystem) loadSnapshots(tenantID, infile string) error {
 		glog.Errorf("Could not load services from %s: %s", label, err)
 		return err
 	}
+
 	// Restore the service data
```
Comment (smousa) [inline dfs/backup.go]: We don't need this anymore, right?

File: dfs/backup.go [hunk 2]
```diff
@@ -547,68 +549,6 @@ func (dfs *DistributedFilesystem) loadSnapshots(tenantID, infile string) error {
 	return nil
 }
 
-func (dfs *DistributedFilesystem) importSnapshots(filename string) error {
```
Comment (smousa) [inline dfs/backup.go]: Why did you delete this function?

File: dfs/snapshot.go [hunk 3]
```diff
@@ -299,7 +299,7 @@ func (dfs *DistributedFilesystem) DeleteSnapshots(tenantID string) error {
 	return nil
 }
 
-func (dfs *DistributedFilesystem) restoreServices(svcs []*service.Service) error {
+func (dfs *DistributedFilesystem) restoreServices(tenantID string, svcs []*service.Service) error {
```
Comment (smousa) [inline dfs/snapshot.go]: Does it make sense to figure out the tenant ID from the list of passed-in services?

File: dfs/snapshot.go [hunk 4]
```diff
@@ -365,28 +365,11 @@ func (dfs *DistributedFilesystem) restoreServices(svcs []*service.Service) error
 					glog.Errorf("Could not add service %s: %s", serviceID, err)
 					return err
 				}
+			}
 
-				/*
-					// TODO: enable this to generate a new service ID, instead of recycling
-					// the old one
```
Comment (smousa) [inline dfs/snapshot.go]: Should we keep this TODO in case we decide we want to do this in the future?

**Comment by smousa** [comment]
Don't forget to create another pull request into develop

File: dfs/backup.go (hunk 2 above)
Comment (jafcrocker) [inline dfs/backup.go]: It was not used and called dfs.Rollback.  I cleaned it up to help simplify my code analysis.

File: dfs/snapshot.go (hunk 3 above)
Comment (jafcrocker) [inline dfs/snapshot.go]: Possibly. given that we know what tenant we are trying to restore it made sense to me to use that information rather than try to infer it.  One could make a case for asserting that the tenant ID inferred from the snapshot matches the expected
